### ISSUE TITLE
bad parameter reference at elimOpaque

### ISSUE THREAD

**Issue by larsrh** [lampepfl/dotty#11265]
## Compiler version

3.0.0-M3

## Minimized code

https://github.com/typelevel/semigroups/tree/wip/dotty

`sbt ++3.0.0-M3! coreJVM/compile`

(Every time I try to minimize it further, I hit #11264)

## Output

```scala
[error] -- Error: /home/lars/proj/typelevel/semigroups/core/src/main/scala/io/chrisdavenport/semigroups/Dual.scala:20:4 
[error] 20 |    new CommutativeMonad[Dual] with NonEmptyTraverse[Dual] with Distributive[Dual]{
[error]    |    ^
[error]    |bad parameter reference io.chrisdavenport.semigroups.Dual#A at elimOpaque
[error]    |the parameter is type A in class Dual but the prefix io.chrisdavenport.semigroups.Dual
[error]    |does not define any corresponding arguments.
```

## Expectation

It should compile.

**Comment by smarter** [comment]
Would you mind trying again with a nightly build? A problem with the same symptoms was recently fixed: https://github.com/lampepfl/dotty/issues/11057

**Comment by larsrh** [comment]
Confirm it is fixed in `3.0.0-RC1-bin-20210130-ab3bc41-NIGHTLY`. Now I just get a bunch of #11264's.
